Minimum golang version for kn

I am running Fedora 29 and golang versions as below
```
13:37 ➜  client git:(fix-docs)  cat /etc/redhat-release 
Fedora release 29 (Twenty Nine)
13:38 ➜  client git:(fix-docs)  rpm -q golang
golang-1.11.11-1.fc29.x86_64
```

latest master fails to build for me as it requires golang 1.12 for strings.ReplaceAll,
```
13:39 ➜  client git:(master)  ./hack/build.sh 
🕸️ Update
🧽 Format
⚖️ License
📖 Docs
# github.com/knative/client/pkg/kn/commands/plugin
pkg/kn/commands/plugin/plugin_verifier.go:280:12: undefined: strings.ReplaceAll
```

Our docs point 1.11.4 or later, that should be replaced with 1.12.7 or latest (the latest one available https://golang.org/dl/)

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -24,7 +24,7 @@ Start by creating [a GitHub account](https://github.com/join), then set up
 You must install these tools:
 
 1. [`go`](https://golang.org/doc/install): The language `Knative Client` is
-   built in (1.11.4 or later)
+   built in (1.12 or later)
 1. [`goimports`](https://godoc.org/golang.org/x/tools/cmd/goimports)
 1. `gcc` compiler: Used during testing. Not needed if golang is installed
    via the installer

